python/tvm/script/node.py

```diff
@@ -133,18 +138,27 @@ def check_index(index: Union[int, PrimExpr]):
     def __str__(self):
         regions: List[str] = []
         for s in self.slices:
-            if s.stop is None:
-                regions.append(str(s.start))
-            else:
-                regions.append(str(s.start) + ": " + str(s.stop))
+            if isinstance(s, Slice):
+                if s.stop is None:
+                    regions.append(str(s.start))
+                else:
+                    regions.append(str(s.start) + ": " + str(s.stop))
+            elif isinstance(s, BufferSlice):
+                regions.append(s.buffer.name + "[" + str(s.start) + "]")
 
         return self.buffer.name + "[" + ", ".join(regions) + "]"
 
     def asobject(self) -> BufferLoad:
         """Convert object."""
+        indices: List[PrimExpr] = []
         for s in self.slices:
-            if s.stop is not None:
-                self.report_error("BufferLoad only accepts elementwise access", self.span)
-
-        indices = [s.start for s in self.slices]
+            if isinstance(s, Slice):
+                if s.stop is not None:
+                    self.report_error("BufferLoad only accepts elementwise access", self.span)
+                indices.append(s.start)
+            elif isinstance(s, BufferSlice):
+                args: List[PrimExpr] = []
+                for idx in s.slices:
+                    args.append(idx.start)
```

Comment:
Why is the conversion from from BufferSlice to BufferLoad necessary here? Plus it seems like this should be handled by the recursive case.